dwipreproc explicit volume recombination

https://github.com/MRtrix3/mrtrix3/blob/16090b5b4ccdad9652bada2473e904ecfc64c047/bin/dwipreproc#L996

translates to

```
mrcalc field_map.nii.gz 0.123 -mult - | mrfilter - gradient - | mrconvert - field_deriv_pe_1.mif -coord 3 1 -axes 0,1,2
```
causing

```
mrcalc: [100%] uncompressing image "field_map.nii.gz"
mrcalc: [100%] computing: (field_map.nii.gz * 0.123)
mrfilter: [100%] applying smoothing prior to calculating gradient
mrfilter: [100%] applying gradient filter to image -
mrconvert: [ERROR] omitted axis "4" has dimension greater than 1
```
when performing explicit volume recombination

field_map.nii.gz is 4D containing 6 volumes
after mrfilter this becomes 5D containing 3 x 6 volumes
subsequent mrconvert then fails when trying to omit a non-singleton dimension

--- a/.travis.yml
+++ b/.travis.yml
@@ -94,7 +94,7 @@ install:
   - |
     if [[ "${test}" == "build" || "${test}" == "run" ]]; then
         export EIGEN_CFLAGS=-I`pwd`/../eigen;
-        (cd ..; hg clone https://bitbucket.org/eigen/eigen/; cd eigen; hg update 3.3);
+        ( cd ..; git clone https://github.com/eigenteam/eigen-git-mirror.git eigen; cd eigen; git checkout branches/3.3 )
     fi
 before_script:
   #######################################################################################################################

--- a/bin/dwipreproc
+++ b/bin/dwipreproc
@@ -964,10 +964,20 @@ else:
   #   (even if this doesn't cause an issue with the subsequent mrcalc command, it may in the future, it's better for
   #   visualising the script temporary files, and it gives the user a warning about an out-of-date FSL)
   field_map_image = fsl.findImage('field_map')
-  if not image.match('topup_in.nii', field_map_image):
+  field_map_header = image.Header(field_map_image)
+  if not image.match('topup_in.nii', field_map_header, 3):
     app.warn('topup output field image has erroneous header; recommend updating FSL to version 5.0.8 or later')
     new_field_map_image = 'field_map_fix.mif'
     run.command('mrtransform ' + field_map_image + ' -replace topup_in.nii ' + new_field_map_image)
+    file.delTemporary(field_map_image)
+    field_map_image = new_field_map_image
+  # In FSL 6.0.0, field map image is erroneously constructed with the same number of volumes as the input image,
+  #   with all but the first volume containing intensity-scaled duplicates of the uncorrected input images
+  # The first volume is however the expected field offset image
+  elif len(field_map_header.size()) == 4:
+    app.console('Correcting erroneous FSL 6.0.0 field map image output')
+    new_field_map_image = 'field_map_fix.mif'
+    run.command('mrconvert ' + field_map_image + ' -coord 3 0 -axes 0,1,2 ' + new_field_map_image)
     file.delTemporary(field_map_image)
     field_map_image = new_field_map_image
   file.delTemporary('topup_in.nii')

